### ISSUE TITLE
[BUG] Imviz: "White out" when center_on "off-screen, on-image"

### ISSUE THREAD

**Issue by pllim** [spacetelescope/jdaviz#704]
As reported by @ojustino while reviewing #687 .

To reproduce:

1. Fire up Imviz.
2. Load some image.
3. Use `imviz.center_on` on a pixel location that is valid but far out from the displayed area.
4. See nothing being displayed and no error. Viewer also becomes non-functional and has to be reloaded.

Example use case from Justin:

```python
from astropy.coordinates import SkyCoord
from astropy.utils.data import download_file

from jdaviz.configs.imviz import Imviz

imviz = Imviz()

cygA1 = download_file('https://hla.stsci.edu/cgi-bin/getdata.cgi?dataset=hst_05368_01_wfpc2_f555w_pc_drz.fits', cache=True)
cygA2 = download_file('https://hla.stsci.edu/cgi-bin/getdata.cgi?dataset=hst_05368_01_wfpc2_f555w_wf_drz.fits', cache=True)

imviz.load_data(cygA1, data_label='cygA1')
imviz.load_data(cygA2, data_label='cygA2')

sky = SkyCoord('09h59m26.02s 40d44m17.92s')
imviz.center_on(sky)
```

[🐱](https://jira.stsci.edu/browse/JDAT-1577)

**Comment by pllim** [comment]
@astrofrog , maybe you know how to force data load when I update the x/y_min/max attributes?

**Comment by pllim** [comment]
Turns out, this is caused by WCS translation silently passing back `(np.array(np.nan), np.array(np.nan))`

**Comment by pllim** [comment]
With #755 , the original example will result in a warning due to invalid WCS -> pix conversion. So, there is no more problem of panning to nowhere.

**Comment by ojustino** [comment]
I checked this again since I was the one who first reported it. Reading more closely, the white out was happening with invalid coordinates -- for example, the right ascension in the provided example is outside of the queried FITS images' fields of view, so it should trigger an out of bounds message.

Besides that, the white out still happens for me. I was previously using Edge, but I still see this behavior while using Firefox with the current `main` build of `jdaviz` that includes #755.

I'm also seeing a recurrence of the problem where I get the "(X, Y) is out of bounds" message for valid pixel or RA/Dec coordinates that are outside of the area currently displayed in the viewer. If I pan to a section of the image that does include those coordinates, then `center_on` works without issue.

**Comment by pllim** [comment]
@ojustino , your original example now emits a out of bounds warning, which I thought was the desired behavior. Could you please open a new issue and provide data and reproducible code snippets for the problems that still occur for you?

When I tested locally, besides your original example, I zoomed in a lot into a section of image, and then `center_on` the entirely different part that was out of display, and that worked for me. 🤔

**Comment by ojustino** [comment]
@pllim I'm using the same snippet as above when I get the white out instead of the out of bounds warning. I was saying that number 3 in the above steps should be "Use `imviz.center_on` on a pixel location that is invalid" because the supplied coordinates are outside of the images' fields of view.

I believe everyone is getting the intended behavior but me, so I'm not sure that the issue is actually with `jdaviz`. That makes me hesitant to open another issue since the problem may just be my setup.

**Comment by pllim** [comment]
To close this out: @ojustino has confirmed that this bug is now gone.

**Comment by stscijgbot-jwql** [comment]
_This issue is tracked on JIRA as [JDAT-1577](https://jira.stsci.edu/browse/JDAT-1577)._

**Comment by stscijgbot-jwql** [comment]
_Comment by [Pey-Lian Lim](https://jira.stsci.edu/secure/ViewProfile.jspa?name=lim) on [JIRA](https://jira.stsci.edu/browse/JDAT-1577?focusedCommentId=542413#comment-542413):_

Fix at #755
